### PR TITLE
security fixes

### PR DESCRIPTION
**Pull request by Shilpa-Manjunath**

### CODE REVIEW

File: admin/admin-service/src/main/java/io/mosip/admin/packetstatusupdater/service/impl/PacketStatusUpdateServiceImpl.java [hunk 1]
```diff
@@ -102,9 +103,9 @@ private PacketStatusUpdateResponseDto getPacketStatus(String rId) {
 
 			HttpHeaders packetHeaders = new HttpHeaders();
 			packetHeaders.setContentType(MediaType.APPLICATION_JSON);
-			StringBuilder urlBuilder = new StringBuilder();
-			urlBuilder.append(packetUpdateStatusUrl).append(SLASH).append(rId);
-			ResponseEntity<String> response = restTemplate.getForEntity(urlBuilder.toString(), String.class);
+			UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromUriString(packetUpdateStatusUrl + SLASH)
```
Comment (ase-101) [inline admin/admin-service/src/main/java/io/mosip/admin/packetstatusupdater/service/impl/PacketStatusUpdateServiceImpl.java:106]: can you pls remove the SLASH concatenation.

Comment (Shilpa-Manjunath) [reply admin/admin-service/src/main/java/io/mosip/admin/packetstatusupdater/service/impl/PacketStatusUpdateServiceImpl.java]: Done with the changes

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/entity/RegistrationCenter.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ public Object clone() {
 		try {
 			return super.clone();
 		} catch (Exception ex) {
-			return null;
+			return "";
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/entity/RegistrationCenter.java:197]: it should return null

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/exception/ApiExceptionHandler.java [hunk 3]
```diff
@@ -194,7 +198,7 @@
 	@ExceptionHandler(value = { Exception.class, RuntimeException.class })
 	public ResponseEntity<ResponseWrapper<ServiceError>> defaultErrorHandler(
 			final HttpServletRequest httpServletRequest, Exception e) throws IOException {
-		e.printStackTrace();
+		logger.info(e.getMessage(),e);
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/exception/ApiExceptionHandler.java:201]: change the logging level to error

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java [hunk 4]
```diff
@@ -261,7 +261,7 @@ public StatusResponseDto deleteDynamicFieldValue(String id) {
 						SchemaErrorCode.DYNAMIC_FIELD_NOT_FOUND_EXCEPTION.getErrorMessage());
 
 			JsonNode valueJson = dynamicField.getValueJson() != null ? objectMapper.readTree(dynamicField.getValueJson()) : null;
-			String code = valueJson.get("code").toString();
+			String code = null!=valueJson ? valueJson.get("code").toString():"";
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java:264]: throw back exception if valueJson is null

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java [hunk 5]
```diff
@@ -351,7 +351,7 @@ public StatusResponseDto updateDynamicFieldValueStatus(String id, boolean isActi
 						SchemaErrorCode.DYNAMIC_FIELD_NOT_FOUND_EXCEPTION.getErrorMessage());
 
 			JsonNode valueJson = dynamicField.getValueJson() != null ? objectMapper.readTree(dynamicField.getValueJson()) : null;
-			String code = valueJson.get("code").toString();
+			String code = null!=valueJson?valueJson.get("code").toString():"";
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java:354]: throw back exception if valueJson is null

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/impl/SyncAuthTokenServiceImpl.java [hunk 6]
```diff
@@ -264,7 +264,7 @@ private void validateRequestTimestamp(LocalDateTime requestTimestamp) {
                 break;
         }
 
-        ResponseWrapper<TokenResponseDto> responseWrapper = objectMapper.readValue(responseEntity.getBody(),
+        ResponseWrapper<TokenResponseDto> responseWrapper = objectMapper.readValue(null!=responseEntity?responseEntity.getBody():"",
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/impl/SyncAuthTokenServiceImpl.java:267]: Not required to handle null check here, this is excepted to be handled by global exceptionHandler

Comment (Shilpa-Manjunath) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/impl/SyncAuthTokenServiceImpl.java:267]: Done

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java (hunk 5 above)
Comment (Shilpa-Manjunath) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java:354]: Done

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java (hunk 4 above)
Comment (Shilpa-Manjunath) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DynamicFieldServiceImpl.java:264]: Done

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/exception/ApiExceptionHandler.java (hunk 3 above)
Comment (Shilpa-Manjunath) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/exception/ApiExceptionHandler.java:201]: Done

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/entity/RegistrationCenter.java (hunk 2 above)
Comment (Shilpa-Manjunath) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/entity/RegistrationCenter.java:197]: Done

**Review by ase-101 — APPROVED** [review]

**Pull request merged**